src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.birth();
+  }
+
+  birth() {
+    Animal.alive.push(this);
+  }
+
+  death() {
+    Animal.alive = Animal.alive.filter(animal => animal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (!victim.hidden && victim.hidden !== undefined) {
```

Comment:
> Also, you should check if victim IS Herbivore, please add this check

please pay attention to comments and add proper check, replace your `undefined` check with `instanceof` for that

Reply:
yes, it's much much better) i've done final version, please check :)